src/Ledger/Ratify.lagda

```diff
@@ -1,5 +1,105 @@
+Governance actions are \defn{ratified} through on-chain voting actions.
+Different kinds of governance actions have different ratification requirements
+but always involve \textit{two of the three} governance bodies, with the exception
+of a hard-fork initiation, which requires ratification by all governance bodies.
+Depending on the type of governance action, an action will thus be ratified when
+a combination of the following occurs:
+\begin{itemize}
+\item
+  the constitutional committee approves of the action (the number of
+  members who vote `yes' meets the threshold of the constitutional
+  committee)
+\item
+  the DReps approve of the action (the stake controlled by the DReps who
+  vote `yes' meets a certain threshold of the total active voting stake)
+\item
+  the SPOs approve of the action (the stake controlled by the SPOs who
+  vote `yes' meets a certain threshold over the total registered voting
+  stake)
+\end{itemize}
+\textbf{Warning}. Different stake distributions apply to DReps and SPOs.
+\\[4pt]
+A successful motion of no-confidence, election of a new constitutional
+committee, a constitutional change or a hard-fork delays ratification of
+all other governance actions until the first epoch after their
+enactment. This gives a new constitutional committee enough time to vote
+on current proposals, re-evaluate existing proposals with respect to a
+new constitution, and ensures that the in principle arbitrary semantic
+changes caused by enacting a hard-fork do not have unintended
+consequences in combination with other actions.
+
+\subsection{Ratification requirements}
+\label{sec:ratification-requirements}
+Figure~\ref{fig:ratification-requirements} details the ratification requirements for each
+governance action scenario. The columns represent:
+\begin{itemize}
+\item
+  \textbf{Governance action type}. The type of governance action. Note
+  that the protocol parameter updates are grouped into three categories.
+\item
+  \textbf{Constitutional committee (CC)}. A value of \ding{51} indicates
+  that the constitutional committee must approve this action.
+  A value of - means that constitutional committee votes do not apply.
+\item
+  \textbf{DReps}. The DRep vote threshold that must be met as a
+  percentage of \emph{active voting stake}. A value of - means that
+  DReps votes do not apply.
+\item
+  \textbf{SPOs}. The SPO vote threshold which must be met as a percentage
+  of the stake held by all stake pools. A value of - means that SPO
+  votes do not apply.
+\end{itemize}
+\begin{figure*}[h!]
+\begin{longtable}[]{@{}
+   >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.65}}
+  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.11}}
+  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.12}}
+  >{\raggedright\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.12}}@{}}
+\textbf{Action}  & \textbf{CC}  &  \textbf{DRep} & \textbf{SPO} \\
+\hline
+\endhead
```

Comment:
Hmmm... now I'm confused.  I thought you said earlier that you wanted the table to have descriptions rather, e.g., the actual constructor, since the latter are evident from the code.  Perhaps that only applied to the table in the GovernanceAction module...? 🤔

Reply:
Is that what you meant by "this should name the actual constructor?" (i.e., what do you mean by "this" ...sorry)